os/env/path-extender-posix/path-extender-posix.ts

```diff
@@ -70,7 +70,9 @@ async function updateShell (
     return setupFishShell(pnpmHomeDir, opts)
   }
   }
-  if (currentShell == null) throw new PnpmError('UNKNOWN_SHELL', 'Could not infer shell type.')
+  if (!currentShell) throw new PnpmError('UNKNOWN_SHELL', 'Could not infer shell type.', {
+    hint: 'Either set the SHELL environment variable or replace "sh" in the installer command with "bash".'
```

Comment:
> or replace "sh" in the installer command with "bash".'

what does this mean?

Does it mean in this script:

```
curl -fsSL https://get.pnpm.io/install.sh | sh -
```

replace to sh?

```
curl -fsSL https://get.pnpm.io/install.sh | bash -
```

Reply:
Yes. Is this part of the message too confusing? Should I remove it?

Reply:
It is confusing right now. Maybe put more details to the docs instead.